lib/ddtrace/contrib/grape/endpoint.rb

```diff
@@ -196,6 +204,18 @@ def analytics_sample_rate
             datadog_configuration[:analytics_sample_rate]
           end
 
+          def handle_4xx_error
+            datadog_configuration[:dont_report_4xx]
+          end
+
+          def exception_is_error?(exception)
```

Comment:
nice work here, I think this handles the logic around  `dont_report_4xx` very thoughtfully. 

That being said, we'll have to update this logic to accomodate `error_statuses`. Apologies for the repeated work here